dask/bag/core.py

```diff
@@ -183,6 +183,42 @@ def finalize_item(results):
     return results[0]
 
 
+def merge_kwargs(dsk, kwargs):
+    """ Merges dask values from kwargs with dsk.
+
+    Returns the merged dsk and a graph expression useful for use with apply in
+    a dask graph to pass the final computed **kwargs to a function.
+
+    Currently only dask.bag.Item and python literal values are supported.
+    """
+    kw_pairs = []
+    for key, val in iteritems(kwargs):
+        if isinstance(val, Item):
+            dsk = merge(dsk, val.dask)
+            val = val.key
+        # TODO elif isinstance(val, Value):
+        elif isinstance(val, Base):
+            raise NotImplementedError(
+                '%s not supported as kwarg value to Bag.map_partitions'
+                % type(val).__name__)
+        kw_pairs.append([key, val])
+    return dsk, kw_pairs
+
+
+def with_kwargs(func, kwargs):
+    """ Wraps func so that it gets called with **kwargs
+
+    Useful for deferring a partial(func, **kwargs) with eventually-computed
+    kwargs in a dask graph.
+
+    TODO: if core supported splicing args for us, we could replace usage of
```

Comment:
Feedback welcome here, this smelled like "too much technology" similarly to #1044 and so I wrote it down as scope creep for this PR.  Either way (be it computed functions or arg splicing) I put it to you that you're missing a crucial part in your "incomplete lisp" around this pain point ;-)